src/compose.ts

```diff
@@ -65,9 +66,28 @@ services:
  * does not have the expected structure and discrepancies can't be resolved,
  * validation errors are thrown. The input composition is mutated in-place.
  *
- * @param c The input composition as a plain JS object
+ * @param o The input composition as a plain JS object
+ * @param fileResolverCb Callback to access filePaths
+ * 	Has to validate that no symbolic links are used outside project folder
  */
-export function normalize(o: any): Composition {
+export function normalize(o: any): Composition;
+export async function normalize(
+	o: any,
+	fileResolverCb: (path: string) => Promise<ReadStream>,
+): Promise<Composition>;
+export function normalize(
+	o: any,
+	fileResolverCb?: (path: string) => Promise<ReadStream>,
+): Composition | Promise<Composition> {
+	if (fileResolverCb === undefined) {
+		return normalizeObjecToComposition(o);
+	} else {
+		const composition = normalizeObjecToComposition(o);
+		return expandContext(composition, fileResolverCb);
+	}
+}
+
+function normalizeObjecToComposition(o: any): Composition {
```

Comment:
`normalizeObjecToComposition` is missing a `t` in `Object`, it was stolen by `To`.